src/coreclr/runtime/amd64/WriteBarriers.S

```diff
@@ -261,6 +269,14 @@ LEAF_END RhpCheckedXchg, _TEXT
 LEAF_ENTRY RhpByRefAssignRef, _TEXT
 ALTERNATE_ENTRY RhpByRefAssignRefAVLocation1
     mov     rcx, [rsi]
+#ifdef TARGET_APPLE
+// Apple's linker has issues which break unwind info if
```

Comment:
What's the manifestation of the bad unwind info?

 If it is just when single stepping in lldb, do we care?

Reply:
The manifestation is that if we AV on this location, we are unable to unwind to the managed frame, so we failfast the process instead of producing a NullReferenceException which can be caught.

Reply:
Would it be better to use the same manual uwnind routing as in regular CoreCLR as we use in NAOT to avoid this problem?

Reply:
There is another approach where we don't use libunwind for unwinding these entries, instead we use our own unwinder which is special cased to know what to do.  (This is what NativeAot does.)

Reply:
(Unlike NativeAOT, CoreCLR implements the Apple compact unwinding through a modified copy of the llvm-libunwind code; libunwind should not even come into play for this since it's covered by the compact codes.)

Reply:
When I was doing fixes of the osx-arm64 NativeAOT port I found a bug in llvm-libunwind comparison of the boundary comparison for IP addresses and function start/end (https://github.com/dotnet/runtime/commit/cb6fb60dfc253bc13655fafa15686cb864fd0ea2#diff-fa7a2ccbf98ccd844bfc21e48f2bc700137c2903cf97d487fa011e16b1f9b3c0).

I looked at the custom unwinding code in CoreCLR and there are similar comparisons that look sketchy:
https://github.com/dotnet/runtime/blob/c201ad1d6bdf9b1a287748b44239c6aef8ecfb72/src/coreclr/pal/src/exception/remote-unwind.cpp#L1034-L1037
https://github.com/dotnet/runtime/blob/c201ad1d6bdf9b1a287748b44239c6aef8ecfb72/src/coreclr/pal/src/exception/remote-unwind.cpp#L1064-L1067
(in both cases the condition should presumably be `ip >= funcEnd`, not `ip > funcEnd`)